js/language_server/Main.ml

```diff
@@ -0,0 +1,105 @@
+(* Austin Theriault
+ *
+ * Copyright (C) 2019-2023 Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(* Commentary *)
+(*  *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+
+open Js_of_ocaml
+open Semgrep_js_shared
+
+let read_line_ref = ref (fun () -> Lwt.return None)
+let read_exactly_ref = ref (fun _ -> Lwt.return None)
```

Comment:
Not sure why these are refs/what they end up getting swapped out with (saw the methods later, but would like more context). Would be good docs to have.

Reply:
Got rid of them